modules/utask/execution.go

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package utask
+
+import (
+	"encoding/gob"
+	"fmt"
+	"reflect"
+	"runtime"
+
+	"github.com/pkg/errors"
+)
+
+var fnNameMap = make(map[string]interface{})
+
+// Signature represents a function and its arguments
+type Signature struct {
+	FnName string
+	Args   []interface{}
+}
+
+// Execute executes the function
+func (s *Signature) Execute() ([]reflect.Value, error) {
+	var targetArgs []reflect.Value
+	for _, arg := range s.Args {
+		targetArgs = append(targetArgs, reflect.ValueOf(arg))
+	}
+	if fn, ok := fnNameMap[s.FnName]; ok {
+		fnValue := reflect.ValueOf(fn)
+		return fnValue.Call(targetArgs), nil
+	}
+	return nil, fmt.Errorf("Function: %s not found. Did you forget to register?", s.FnName)
```

Comment:
nit: errors should start with lowercase. also use `%q` for function name to auto-surround with quotes so it's clear where the name starts and ends